mon-pix/app/services/session.js

```diff
@@ -64,7 +64,8 @@ export default class CurrentSessionService extends SessionService {
   }
 
   async handleUserLanguageAndLocale(transition = null) {
-    await this._loadCurrentUserAndSetLocale(transition?.to?.queryParams?.lang);
+    const language = this.locale.handleSupportedLanguage(transition?.to?.queryParams?.lang);
+    await this._loadCurrentUserAndSetLocale(language);
```

Comment:
Pourquoi transformer une locale fournie invalide au lieu de simplement l'ignorer ?

Par exemple dans Pix Orga on ignore simplement les demandes de locale invalide : https://github.com/1024pix/pix/blob/dev/orga/app/services/session.js#L34

```suggestion
    if (!this.locale.isSupportedLanguage(transition?.to?.queryParams?.lang) {
      return;
    }
    await this._loadCurrentUserAndSetLocale(language);
```

Reply:
Je dirais de regarder ça à tête reposée, ce code est complexe, je serais d'avis de s'y pencher quand on traitera ce ticket https://1024pix.atlassian.net/browse/PIX-7570
Je note ta remarque dans le ticket

Dans la fonction _loadCurrentUserAndSetLocale y'a des cas ou on fait un if/else sur le fait qu'il y ait des query params ou non, si on lui passe toujours language, on rentrera plus dans ces cas là. J'ai aucune idée là maintenant des conséquences. D'où mes réserves

Reply:
👌 bien compris et d'accord ! Merci !